service/service.go

```diff
@@ -39,33 +42,41 @@ func GetServer(addr string, n http.Handler) *http.Server {
 }
 
 func GetRouter(as *APIServer, uis *UIServer) (http.Handler, error) {
+	app := gimlet.NewApp()
+	app.ResetMiddleware()
+	app.AddMiddleware(NewRecoveryLogger())
+	app.AddMiddleware(gimlet.UserMiddleware(uis.UserManager, GetUserMiddlewareConf()))
+	app.AddMiddleware(negroni.NewStatic(http.Dir(filepath.Join(uis.Home, "public"))))
+
 	// in the future, we'll make the gimlet app here, but we
 	// need/want to access and construct it separately.
-	app, err := GetRESTv1App(as, uis.UserManager)
+	restv1, err := GetRESTv1App(as)
 	if err != nil {
 		return nil, err
 	}
 
-	app.AddMiddleware(negroni.NewStatic(http.Dir(filepath.Join(uis.Home, "public"))))
+	restv2API := gimlet.NewApp()
+	restv2UI := gimlet.NewApp()
+	restv2UI.ResetMiddleware()
+
+	restv2API.SetPrefix(evergreen.APIRoutePrefix + "/" + evergreen.RestRoutePrefix)
+	restv2UI.SetPrefix(evergreen.RestRoutePrefix)
+
+	// attach restv2 handlers
+	route.AttachHandler(restv2API, as.queue, as.Settings.ApiUrl, as.Settings.SuperUsers, []byte(as.Settings.Api.GithubWebhookSecret))
+	route.AttachHandler(restv2UI, uis.queue, uis.Settings.Ui.Url, uis.Settings.SuperUsers, []byte(uis.Settings.Api.GithubWebhookSecret))
```

Comment:
Don't we need to attach the restV2 routes to the UI endpoint as well?

Reply:
yeah, there's really only one endpoint (they both have all of the same routes) but rest v2 (like restv1 is accessible at both /api/rest/v2/ and /rest/v2/

it looks like we used to put restv1 in both places but we don't any more. I'd be willing to take restv2 out of the API later (with the /api prefix).